acceptance/tests/failover/timeout_failover.rb

```diff
@@ -0,0 +1,44 @@
+require 'pxp-agent/config_helper.rb'
+require 'pxp-agent/test_helper.rb'
+
+test_name 'C97964 - agent should use next broker if primary is timing out' do
+
+  blocked_broker_instance = 0
+  teardown do
+    unblock_pcp_broker(master,blocked_broker_instance)
+    broker_instance = 0 # 0 indexed
+    kill_pcp_broker(master)
+    run_pcp_broker(master, broker_instance)
+  end
+
+  step 'Ensure each agent host has pxp-agent configured with multiple uris, running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      num_brokers = 2
+      pxp_config = pxp_config_hash_using_puppet_certs(master, agent, num_brokers)
+      pxp_config['allowed-keepalive-timeouts'] = 0
+      create_remote_file(agent, pxp_agent_config_file(agent), pxp_config.to_json.to_s)
+      on(agent, "rm -rf #{logfile(agent)}")
+      on agent, puppet('resource service pxp-agent ensure=running')
+      show_pcp_logs_on_failure do
+        assert(is_associated?(master, "pcp://#{agent}/agent"),
```

Comment:
is_associated ends up doing an inventory request, which finds a broker uri, which can only send the request to the correct broker.  so this is sure to show that we are talking to the primary/correct broker.  same as in the reconnect test.  

we can't assert anything on the replica broker, because it's not running at this point.  i suppose it could be.